diff --git a/setup.py b/setup.py
index c8312b8..602d40b 100644
--- a/setup.py
+++ b/setup.py
@@ -139,7 +139,7 @@ def classy_extension_config():
     config['extra_compile_args'] = []
     # important or get a symbol not found error, because class is
     # compiled with c++?
-    config['language'] = 'c'
+    config['language'] = 'c++'
     config['libraries'] = ['class', 'gfortran', 'm']
 
     # determine if swig needs to be called
@@ -161,7 +161,7 @@ if __name__ == '__main__':
           extras_require={'tests': ['runtests', 'astropy', 'scipy']},
           ext_modules = cythonize([
                         Extension(**classy_extension_config())
-          ]),
+          ], language_level = '2'),
           libraries=[libclass_config()],
           cmdclass = {
               'sdist': custom_sdist,
